test/automation/src/extensions.ts

```diff
@@ -72,6 +72,25 @@ export class Extensions extends Viewlet {
 			await this.code.waitForElement(`.extension-editor .monaco-action-bar .action-item:not(.disabled) a[aria-label="Disable this extension"]`);
 		}
 	}
+
+	private async waitForExtensionToBeInstalled(): Promise<void> {
+		let attempt = 1;
+		while (true) {
+			try {
+				await this.code.waitForElement(`.extension-editor .monaco-action-bar .action-item:not(.disabled) .extension-action.uninstall`, undefined);
+				break;
+			} catch (err) {
+				if (await this.code.getElement(`.extension-editor .monaco-action-bar .action-item .extension-action.install.installing`)) {
+					if (attempt++ === 3) {
```

Comment:
There already is a `if (attempt++ === 3) {` in the calling code though :thinking:

Reply:
Yeah, they are two different attempts. New one is to wait longer if an extension installation goes for long time.